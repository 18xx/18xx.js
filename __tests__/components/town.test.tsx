import { List } from 'immutable';
import * as React from 'react';
import { ReactElement } from 'react';
import * as renderer from 'react-test-renderer';

import Town from '../../src/components/town';

import Point from '../../src/point';

describe('Town', () => {
  describe('#toString', () => {
    it('returns the svg for a circle centered on the point', () => {
      const point: Point = new Point(4, 6);
      const points: List<Point> = List<Point>([point]);
      expect(renderer.create(<Town points={points} />)).toMatchSnapshot();
    });
  });
});
