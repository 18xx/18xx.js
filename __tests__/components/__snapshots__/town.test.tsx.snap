exports[`Town #toString returns the svg for a circle centered on the point 1`] = `
<g>
  <circle
    cx={4}
    cy={6}
    fill="black"
    r={10} />
</g>
`;
